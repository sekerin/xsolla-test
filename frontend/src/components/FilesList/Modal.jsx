import React, { Component } from 'react';
import PropTypes            from 'prop-types';
import { connect }          from 'react-redux';
import {
  Button,
  Modal,
  ModalBody,
  ModalFooter,
  ModalHeader,
  ModalTitle,
  Alert,
  FormGroup,
  ControlLabel,
  FormControl
}                           from 'react-bootstrap';
import ButtonLoader         from 'react-bootstrap-button-loader';

const propsTypes = {
  dispatch: PropTypes.func.isRequired,
  modal: PropTypes.bool,
  value: PropTypes.any,
  error: PropTypes.array,
  item: PropTypes.any,
  loading: PropTypes.bool,
  onClickAdd: PropTypes.func.isRequired,
  onClickUpdate: PropTypes.func.isRequired,
  onClickClose: PropTypes.func.isRequired,
  onChangeFile: PropTypes.func.isRequired
};

class SimpleModal extends Component {
  constructor() {
    super();

    this.handleChange = this.handleChange.bind(this);
    this.handleClickClose = this.handleClickClose.bind(this);
    this.handleClickAdd = this.handleClickAdd.bind(this);
    this.handleClickUpdate = this.handleClickUpdate.bind(this);
  }

  handleChange(e) {
    this.props.onChangeFile(e.target.files[0]);
  }

  handleClickClose() {
    this.props.onClickClose();
  }

  handleClickAdd() {
    if (this.props.value) {
      this.props.onClickAdd(this.props.value);
    }
  }

  handleClickUpdate() {
    if (this.props.value) {
      this.props.onClickUpdate(this.props.item.name, this.props.value);
    }
  }

  render() {
    return (
      <Modal show={this.props.modal}>
        <ModalHeader>
          <ModalTitle>{this.props.item ? this.props.item.name : 'Add name'}</ModalTitle>
        </ModalHeader>
        {this.props.error && (
          <Alert bsStyle='danger'>
            {this.props.error}
          </Alert>
        )}
        <ModalBody>
          <form>
            <FormGroup>
              <ControlLabel>File: {this.props.value.name}</ControlLabel>
              <FormControl
                onChange={this.handleChange}
                type='file'
              />
              <FormControl.Feedback/>
            </FormGroup>
          </form>
        </ModalBody>
        <ModalFooter>
          <Button onClick={this.handleClickClose}>Close</Button>
          <ButtonLoader
            bsStyle='primary'
            loading={this.props.loading}
            onClick={this.props.item ? this.handleClickUpdate : this.handleClickAdd}
          >Save</ButtonLoader>
        </ModalFooter>
      </Modal>
    );
  }
}

SimpleModal.propTypes = propsTypes;

function mapStateToProps(state) {
  const { modal, value, error, item, loading } = state.modal;

  return { modal, value, error, item, loading };
}

export default connect(mapStateToProps)(SimpleModal);
