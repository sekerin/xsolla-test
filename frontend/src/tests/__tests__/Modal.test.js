import React            from 'react';
import { StaticRouter } from 'react-router-dom';

import { shallow }       from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Modal          from '../../components/FilesList/Modal';
import configureStore from '../../redux/configureStore';

describe('Test Modal component', () => {
  it('Empty Modal', () => {
    const store = configureStore({});
    const wrapper = shallow(
      <Modal
        store={store}
        onClickAdd={jest.fn()}
        onClickUpdate={jest.fn()}
        onClickClose={jest.fn()}
        onChangeFile={jest.fn()}
      />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
