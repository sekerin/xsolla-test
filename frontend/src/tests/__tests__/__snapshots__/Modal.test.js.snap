// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Modal component Empty Modal 1`] = `
<SimpleModal
  dispatch={[Function]}
  error={null}
  item={null}
  loading={false}
  modal={false}
  onChangeFile={[MockFunction]}
  onClickAdd={[MockFunction]}
  onClickClose={[MockFunction]}
  onClickUpdate={[MockFunction]}
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "liftedStore": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "liftedStore": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "unsubscribe": [Function],
    }
  }
  value=""
/>
`;
